### DISCUSSION TITLE
Backup of Sessions, Subscriptions and Messages in free version

### DISCUSSION THREAD

**Discussion by on-delete** [emqx/emqx#6333] [Q&A] [answered]
As per the documentation, i understood that all sessions, subscriptions and messages will be backed up into e.g. a database and that the broker instances are stateless. And that this is a enterprise only feature, so we cannot use it in our free version.

Will this behaviour changed with the new EMQX 5 version? Will this become free, or is this a feature that will still be in the enterprise edition only? We need to persist sessions over a broker restart so that no message get lost on these occasions.

**Comment by zmstone** [comment, marked as answer]
Yes. There is going to be a on-disk persistent session support in version 5

**Reply by on-delete** [reply]
Thanks for the quick reply. And will this be a enterprise only feature, or also for the basic version?

**Reply by zmstone** [reply]
it will be a free (of charge) feature for the community edition.

**Reply by xzycn** [reply]
@zmstone hi,it includes messages persistence in version 5 or only sessions?

**Reply by thalesmg** [reply]
Yes, messages will persisted as well while clients with persistent sessions are offline.

**Reply by xzycn** [reply]
@thalesmg cool, thanks for your quick reply.
BTW, when will version 5 come out? I could hardly wait to see it :)

**Reply by wodomierz** [reply]
Doesn't the retainer plugin solve the part of the problem?
https://github.com/emqx/emqx/tree/main-v4.3/apps/emqx_retainer ?

**Reply by zmstone** [reply]
a 'retained' message is a message which is flagged by the publisher to be stored in the broker so the subscribers can receive it immediately after subscribed, instead of waiting for the publisher to publish a new message

it is useful for publishers to announce the 'latest' value for a given topic,
it only stores one (the last) message per topic

**Reply by wodomierz** [reply]
@zmstone Thanks. The sessions aren't stored, but can replication between nodes save the data from the qos process? So in the case when one broker is dead, another one could continue interrupted qos process? As far as I understand your docs, the answer is no, but I could't find the clear answer.

**Reply by zmstone** [reply]
So far, sessions do not survive node restarts.

In 5.0 we'll make EMQ X configurable to make sessions survive node restarts (that is, sessions are replicated).

**Reply by xzycn** [reply]
@zmstone Hi，do you have a plan to release version 5.0? It could  come out in 1~2 months?

**Reply by zmstone** [reply]
5.0 is in beta phase, but it’s hard to say when it will be ready for the final release of v5.0.0

**Reply by xzycn** [reply]
@zmstone well,thank you for your information :)

**Reply by xzycn** [reply]
@zmstone 
Hi, the v5.0 has been released, congrats :)
I can't find  persistent support,do I miss something?
